python/tk_maya/panel_generation.py

```diff
@@ -267,8 +267,13 @@ def build_workspace_control_ui(shotgun_panel_name):
 
     # In the context of this function, we know that we are running in Maya 2017 and later
     # with the newer versions of PySide and shiboken.
-    from PySide2 import QtWidgets
-    from shiboken2 import wrapInstance
+    # Try importing PySide6 modules first
+    try:
+        from PySide6 import QtWidgets
```

Comment:
@staceyoue what do you think about my other idea using the Maya version to decide which PySide to import? (we know that <=2014: Qt4, 2016:2024: Qt5, >=2025: Qt6)

Reply:
Also, adding to Stacey's response to Julien's question...
> I wonder if there is a reason why we don't use tk-core's Qt importer on these engines.

I've seen this imports are then patched and assigned to tank.platform.qt at [tk-core/python/tank/platform /engine.py](https://github.com/shotgunsoftware/tk-core/blob/1e48ecbf0cc7b275688b60eddd7f1b1546397b1c/python/tank/platform/engine.py#L176), just after the Maya engine is initialized.

Reply:
@staceyoue do you mean it is possible to use a TK engine outside Maya? Like for other TK apps, I understand. But for engines, I don't see how it would work or be any useful.

Anyway, I kind of like this Maya version idea but I'm not stuck with it. @NorberMV do what you think is best :)